### PR TITLE
feat: this add api for account-creation

### PR DESCRIPTION
**Pull request by vnko1**

### CODE REVIEW

File: src/components/Fields/PasswordField/PasswordField.tsx [hunk 1]
```diff
@@ -23,7 +23,9 @@ const PasswordField: FC<PasswordFieldProps> = ({
   defaultValue,
   dataAutomation,
   strength,
+  strengthMessage = "Password must be at least 8 characters long, including uppercase letters and special characters",
```
Comment (ArtemBohak) [inline src/components/Fields/PasswordField/PasswordField.tsx:26]: I think we have changed the validation rules, as far as I know, Sasha must have re-written them

Comment (vnko1) [reply src/components/Fields/PasswordField/PasswordField.tsx]: fixed

File: src/containers/App/App.tsx [hunk 2]
```diff
@@ -1,6 +1,19 @@
+import { useAppDispatch } from "@/src/redux/hooks";
 import { Router } from "..";
+import { useEffect } from "react";
+import { getTokenFromLC } from "@/src/utils";
+import { fetchIsAuthUser } from "@/src/redux/slices";
 
 const App = () => {
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
```
Comment (ArtemBohak) [inline src/containers/App/App.tsx:10]: This part will be executed on the fetch of /me data via request and response interceptors on the initial render

File: src/containers/Router/Routes/PrivateRoute/PrivateRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { IRoutesProps } from "../types/Routes.type";
+import { useAppSelector } from "@/src/redux/hooks";
+import { getTokenFromLC, links } from "@/src/utils";
+
+const PrivateRoute: FC<IRoutesProps> = ({
+  component: Component,
+  redirectUrl = links.LOG_IN,
+}) => {
+  const { isAuth } = useAppSelector((state) => state.userSlice);
+  if (!getTokenFromLC()) return <Navigate to="/" replace />;
```
Comment (ArtemBohak) [inline src/containers/Router/Routes/PrivateRoute/PrivateRoute.tsx:13]: The access token can be outdated while there might be a valid refresh token. In the case the access token is invalid, we should try to refresh it. When the fetchIsAuthUser is executed, the axios interceptors will try to pass an access token into Authorization header and if any problems occur, they will try to refresh tokens and the user will be authorized. That is why we should check if isAuth is true. But I think you might have the reasons to define it so and I might have overlooked something

**Review by ArtemBohak — APPROVED** [review]

**Review by BogAlex4ndr — APPROVED** [review]

**Pull request merged**